components/attendance/AttendanceSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import cn from "@/utils/helpers/cn";
+import {
+  AbsenceType,
+  AttendanceEvent,
+  StudentAttendance,
+} from "@/utils/types/attendance";
+import { StylableFC } from "@/utils/types/common";
+import { ChipSet, InputChip } from "@suankularb-components/react";
+import { useTranslation } from "next-i18next";
+
+const AttendanceSelector: StylableFC<{
+  attendance: StudentAttendance;
+  shownEvent: AttendanceEvent;
+  setAttendanceOfShownEvent: (
+    eventAttendance: StudentAttendance[AttendanceEvent],
+    options?: Partial<{ noSave: boolean }>,
+  ) => void;
```

Comment:
1. Usually, when we pass in callbacks that are triggered from a change of state within the child component, we start the callback prop name with `on`, i.e. `onChange` or `onSubmit`. In this context, `onChange` would work best.
2. `options.noSave` doesn’t seem to be used in this component at all. Just remove it from the type declaration.

Actually, Absence Type Selector did this pretty well, check it out:
https://github.com/suankularb-wittayalai-school/mysk-frontend/blob/e0a5ffb94c45ed45cab2fb851bce970b176028d6/components/attendance/AbsenceTypeSelector.tsx#L16-L25